src/components/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react'
+
+interface ErrorBoundaryState {
+  hasError: boolean
+  errorData: {
+    error: null | unknown
+    errorInfo: null | { componentStack: string }
+  }
+}
+
+interface ErrorBoundaryProps {
+  children: React.ReactElement
+}
+
+class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props)
+    this.state = {
+      hasError: false,
+      errorData: {
+        error: null,
+        errorInfo: null,
+      }
+    }
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true }
+  }
+
+  componentDidCatch(error: any, errorInfo: { componentStack: string }) {
+    console.log(error, errorInfo)
+    this.setState({ errorData: { error, errorInfo } })
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <div style={{ paddingLeft: '1rem' }}>
+        <h1>Something went wrong. We are sowwy</h1>
+      </div>
+    }
+
+    return this.props.children || <>Something terrible happened</>
```

Comment:
No need for fallback here. You either show error message or children that were passed to component